src/transformers/models/textnet/modeling_textnet.py

```diff
@@ -0,0 +1,447 @@
+# coding=utf-8
+# Copyright 2023 the Fast authors and HuggingFace Inc. team.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch TextNet model."""
+from typing import Any, List, Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
+from torch import Tensor
+from torch.nn import BCEWithLogitsLoss, CrossEntropyLoss, MSELoss
+
+from transformers import PreTrainedModel, add_start_docstrings
+from transformers.activations import ACT2CLS
+from transformers.modeling_outputs import (
+    BackboneOutput,
+    BaseModelOutputWithPoolingAndNoAttention,
+    ImageClassifierOutputWithNoAttention,
+)
+from transformers.models.textnet.configuration_textnet import TextNetConfig
+from transformers.utils import (
+    add_code_sample_docstrings,
+    add_start_docstrings_to_model_forward,
+    logging,
+    replace_return_docstrings,
+)
+from transformers.utils.backbone_utils import BackboneMixin
+
+
+logger = logging.get_logger(__name__)
+
+# General docstring
+_CONFIG_FOR_DOC = "TextNetConfig"
+_CHECKPOINT_FOR_DOC = "Raghavan/textnet-base"
+_EXPECTED_OUTPUT_SHAPE = [1, 512, 20, 20]
+
+# Image classification docstring
+_IMAGE_CLASS_CHECKPOINT = "Raghavan/textnet-base"
+_IMAGE_CLASS_EXPECTED_OUTPUT = "LABEL_1"
```

Comment:
is there a reason the `model.config.id2label` is not being set for the image classification model?

Reply:
The textnet pretrained weights does not have a classification head.

Reply:
Ok makes sense, comment can be closed